relentless/optimize/objective.py

```diff
@@ -165,3 +168,55 @@ def design_variables(self):
         :class:`ObjectiveFunction` for which the result was constructed, mapped
         to the value of the variables at the time the result was constructed."""
         return self._design_variables
+
+class RelativeEntropy(ObjectiveFunction):
+    def __init__(self, target, simulation, potentials, thermo, dr, communicator=None):
+        self.target = target
+        self.simulation = simulation
+        self.potentials = potentials
+        self.thermo = thermo
+        self.dr = dr
+        self.communicator = communicator
+
+    def compute(self, directory=None):
+        sim = self.simulation.run(self.target, self.potentials, directory, self.communicator)
+        sim_ens = self.thermo.extract_ensemble(sim)
+
+        # compute the relative entropy gradient by integration
+        g_tgt = self.target.rdf
+        g_sim = sim_ens.rdf
+        gradient = {}
+        dvars = self.design_variables()
+
+        for x in dvars:
+            update = 0
+            for i,j in self.potentials.pair.potentials.coeff.pairs:
+                # evaluate derivative wrt x and interpolate
+                dudx = _math.Interpolator(self.potentials.pair.x, self.potentials.pair.derivative((i,j), x))
+
+                # find common domain to compare rdfs
+                r0 = max(g_sim[i,j].table[0][0], g_tgt[i,j].table[0][0])
+                r1 = min(g_sim[i,j].table[-1][0], g_tgt[i,j].table[-1][0])
+                r = np.arange(r0, r1+0.5*self.dr, self.dr)
+
+                # take integral by trapezoidal rule
+                sim_factor = sim_ens.N[i]*sim_ens.N[j]/sim_ens.V.volume
+                tgt_factor = self.target.N[i]*self.target.N[j]/self.target.V.volume
+                mult = 2 if i == j else 4 # 2 if same, otherwise need i,j and j,i contribution
```

Comment:
Just want to make sure: the loop over `self.potentials.pair.potentials.coeff.pairs` (or `self.target.pairs` like Mike suggests) does not double count pairs right? So if we have type A and B particles, we get three pairs (A,A), (A,B), and (B,A)?

Also small nit, maybe move a factor of 2 into the next line and make this statement pick between 1 and 2. It'll be more obvious that `mult` is referring to counting pairs that way.

Reply:
Yep, pairs are the _unique_ pairs of the system: (A,A), (A,B), and (B,B).